accepted/00000-replace-salt-toaster.md

```diff
@@ -0,0 +1,220 @@
+dependencies and use a number of workarounds to still run a test suite.
+
+At the same time, Salt upstream improved their test suite a lot. They added new
+test types and now have a similar test framework to Salt Toaster. 
+
+The purpose of this change is to lessen our maintenance burden and to be good
+open-source citizens that collaborate with our upstream project when it comes to
+testing Salt. Additionally, we have knowledge about running a test suite within
+our team that we want to take advantage of.
+
+# Detailed design
+[design]: #detailed-design
+
+<!-- This is the bulk of the RFC. Explain the design in enough detail for
+somebody familiar with the product to understand, and for somebody familiar with
+the internals to implement. -->
+
+<!-- This section should cover architecture aspects and the rationale behind
+disruptive technical decisions (when applicable), as well as corner-cases and
+warnings. Whenever the new feature creates new user interactions, this section
+should include examples of how the feature will be used. -->
+
+## Overview
+
+The Salt upstream test suite is executed in virtual machines (VMs) that run the
+different operating systems we support. These VMs are set up with sumaform.
+Terracumber triggers terraform to create the VMs from within a Jenkins job and
+launches the test suite, connecting to each VM over SSH. Salt and the test suite
+are installed in the VM via a Salt Bundle that includes the tests.
+
+Pull Request tests are handled slightly different, using a Container containing
+the test dependencies. Management of the dependencies is not done through a
+Dockerfile that lists all dependencies, instead the dependencies are pulled in
+via RPM dependencies.
+
+## Provisioning Test VMs with Sumaform
+
+A new module "salt\_testenv" is added to sumaform. This module defines a new
+role, "salt\_testenv", and the default values needed to run all test suites,
+e.g. 4GiB of RAM. Sumaform installs the test suite from an RPM/Deb package.
+This package is, in most cases, a special Salt Bundle flavor that includes the
+test suite and its dependencies. Only for SLE 15, we additionally use a Salt
+sub-package that only contains the test suite. This sub-package pulls in the
+test dependencies via RPM dependencies.
+
+## Provisioning Test VMs for GitHub Actions
+
+GitHub Actions use so-called runners, which are VMs running Windows, macOS or
+Linux. It is possible to host your own runners, but we currently neither do not
+do that. We can't use sumaform to provision GitHub-hosted runners, so GH Actions
+will use a different approach.
+
+Since the GitHub-hosted Linux runners only run Ubuntu, we will use a Docker
+container that is based on openSUSE Leap (latest version). This container's
+Dockerfile installs a Salt sub-package that pulls in all required test
+dependencies. The container image can be build in OBS, which will automatically
+rebuilt it when included packages change.
```

Comment:
Open question: How will we wait for the built container image with the new code? If everything comes from OBS (code + test-code). Or do we plan on not building this as an RPM on the GH pipeline?

Reply:
For Pull Requests, I think we only need to install the dependencies (`pytest`, `pytest` plugins, 3rd-party libs used in testing etc.) and both the application and test code is "installed" with `actions/checkout`. I don't expect this container to change often and we don't run such tests a lot either. Therefore I didn't consider that we might run into "tests fail due to outdated container".

Reply:
Okay that would be fine by me to remove that additional step of package the test code and the application code on GH.